independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/ResteasyReactiveOutputStream.java

```diff
@@ -226,10 +226,12 @@ private void prepareWrite(ByteBuf buffer, boolean finished) throws IOException {
         if (!committed) {
             committed = true;
             if (finished) {
-                if (buffer == null) {
-                    context.serverResponse().setResponseHeader(HttpHeaderNames.CONTENT_LENGTH, "0");
-                } else {
-                    context.serverResponse().setResponseHeader(HttpHeaderNames.CONTENT_LENGTH, "" + buffer.readableBytes());
+                if (context.serverResponse().getResponseHeader(HttpHeaderNames.CONTENT_LENGTH.toString()) == null) {
```

Comment:
Have you tried with `if (context.serverResponse().headWritten())` instead?

Reply:
Nope. Can try now

Reply:
Yes that works and is neater. Tx